File: packages/backend/src/features/game/game.mapper.ts
import { GameDto } from '~/src/features/game/game.dto';
import { GameModel } from '~/src/features/game/game.model';

export class GameMapper {
  static toDto(game: GameModel): GameDto {
    return {
      id: game.id,
      createdAt: game.createdAt,
      updatedAt: game.updatedAt,
      state: game.state,
      title: game.title,
      description: game.description,
      questions: game.questions,
    };
  }
}

File: packages/backend/src/main/kotlin/quizzem/backend/features/game/mapper/GameMapper.kt
package quizzem.backend.features.game.mapper

import quizzem.backend.features.game.api.dto.GameDto
import quizzem.backend.features.game.model.GameModel

fun GameModel.toDto(): GameDto {
    return GameDto(
        id = id,
        createdAt = createdAt,
        updatedAt = updatedAt,
        state = state,
        title = title,
        description = description,
        questions = emptyList()
    )
}

